src/core/algorithms/fd/tane/tane_common.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "algorithms/fd/pli_based_fd_algorithm.h"
+#include "algorithms/fd/tane/model/lattice_level.h"
+#include "config/error/type.h"
+#include "model/table/column_data.h"
+#include "model/table/column_layout_relation_data.h"
+#include "model/table/position_list_index.h"
+
+namespace algos::tane {
+
+class TaneCommon : public PliBasedFDAlgorithm {
+protected:
+    config::ErrorType max_fd_error_;
+    config::ErrorType max_ucc_error_;
+
+private:
+    void ResetStateFd() final {}
+
+    void Prune(model::LatticeLevel* level);
+    void ComputeDependencies(model::LatticeLevel* level);
+    unsigned long long ExecuteInternal() final;
+    virtual config::ErrorType CalculateZeroAryFdError(ColumnData const* rhs) = 0;
+    virtual config::ErrorType CalculateFdError(model::PositionListIndex const* lhs_pli,
+                                               model::PositionListIndex const* joint_pli) = 0;
+    void RegisterAndCountFd(Vertical const& lhs, Column const* rhs);
+
+public:
+    TaneCommon(std::optional<ColumnLayoutRelationDataManager> relation_manager = std::nullopt);
+    static double CalculateUccError(model::PositionListIndex const* pli,
+                                    ColumnLayoutRelationData const* relation_data);
```

Comment:
Wow, I've never noticed that there (in the original `class Tane`) are such strange architectural decisions in the master branch now (namely, strange public fields and methods).

It seems to me that this public method should also be protected or even private if it is not planned to be used directly in the inherited classes.